lib/spack/spack/version/version_types.py

```diff
@@ -135,79 +194,79 @@ class StandardVersion(ConcreteVersion):
 
     __slots__ = ["version", "string", "separators"]
 
-    def __init__(self, string: Optional[str], version: Tuple[tuple, tuple], separators: tuple):
+    def __init__(self, string: str, version: Tuple[tuple, tuple], separators: tuple):
```

Comment:
Should we be more specific, and say which types the `tuple`s expect?

Reply:
Let me know if you want to do something about `Tuple[tuple, tuple]` in this PR.

Reply:
done in [35fddef](https://github.com/spack/spack/pull/47781/commits/35fddef6837b6799948ea5d15d6d7375278d04ea)